File: core/src/main/java/com/ariasaproject/advancerofrpg/physics/box2d/ContactImpulse.java
package com.ariasaproject.advancerofrpg.physics.box2d;

public class ContactImpulse {
    static {
        initialize();
    }

    final World world;
    long addr;
    float[] tmp = new float[2];

    protected ContactImpulse(World world, long addr) {
        this.world = world;
        this.addr = addr;
    }

    private static native void initialize();

    public native float[] getNormalImpulses();

    public native float[] getTangentImpulses();

    public native int getCount();
}

File: core/src/main/cpp/box2d/ContactImpulse.cpp
#include "ContactImpulse.h"
#include "Dynamics/b2WorldCallbacks.h"

ContactImpulse_M(void, initialize) (JNIEnv
*env,
jclass clazz
)
{
//initialize field ID of JNI
contactImpulsePtr = env->GetFieldID(clazz, "addr", "J");
}
ContactImpulse_M(jfloatArray, getNormalImpulses)(JNIEnv * env, jobject
obj)
{
jfloatArray result = env->NewFloatArray(2);
env->
SetFloatArrayRegion(result,
0, 2, ((b2ContactImpulse*)env->
GetLongField(obj, contactImpulsePtr
))->normalImpulses);
return
result;
}
ContactImpulse_M(jfloatArray, getTangentImpulses)(JNIEnv * env, jobject
obj)
{
jfloatArray result = env->NewFloatArray(2);
env->
SetFloatArrayRegion(result,
0, 2, ((b2ContactImpulse*)env->
GetLongField(obj, contactImpulsePtr
))->tangentImpulses);
return
result;
}
ContactImpulse_M(jint, getCount)(JNIEnv * env, jobject
obj)
{
return ((b2ContactImpulse*)env->
GetLongField(obj, contactImpulsePtr
))->
count;
}